### DISCUSSION TITLE
fails test connection due gocd wanting to use old 2021 github authentication method

### DISCUSSION THREAD

**Discussion by lonkous** [gocd/gocd#12055] [Q&A]
##### Summary

I am trying to setup gocd with git and have provided the right github url however gocd is forcing me to use the email and password even though i am already logged in on the machine and username/password login hasnt  been allowed on github since 2021. This is also my first time using gocd so sorry if i made a basic mistake.

###### Basic environment details

- Go Version: 23.3.0
- Java: openjdk version "17.0.8.1" 2023-08-24
- Linux:  22.04.3 LTS
- kernel: 5.15.0-79-generic

##### Steps to Reproduce

1. create a pipline
2. select git 
3. test connection

##### Expected Results

some form of confirmation that it works

##### Actual Results

Error performing command: --- Command ---
git ls-remote -- https://github.com/user/repo.git refs/heads/dev
--- Environment ---
{}
--- INPUT ----

--- EXIT CODE (128) ---
--- STANDARD OUT ---

--- STANDARD ERR ---
STDERR: fatal: could not read Username for 'https://github.com': No such device or address
---

or when i use the old method of login that does work anymore 

--- STANDARD ERR ---
STDERR: remote: Support for password authentication was removed on August 13, 2021.
STDERR: remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
STDERR: fatal: Authentication failed for 'https://github.com/Abilium-Odoo/oesch1898.git/'

**Comment by chadlwilson** [comment]
Assuming you want to use https rather than ssh, instead of using a direct password for the account, generate a personal access token (classic or fine-grained) which has read access to the required repositories and use that as the 'password'.
